ksp_plugin_adapter/ksp_plugin_adapter.cs

```diff
@@ -1785,24 +1794,69 @@ private enum UnityLayers {
       vessel.mapObject.uiNode.OnClick += OnVesselNodeClick;
       RemoveStockTrajectoriesIfNeeded(vessel);
     }
-    Vessel main_vessel = PredictedVessel();
-    if (main_vessel == null) {
-      return;
+    string main_vessel_guid = PredictedVessel()?.id.ToString();
+    if (main_vessel_guid != null && !plugin_.HasVessel(main_vessel_guid)) {
+      main_vessel_guid = null;
     }
-    string main_vessel_guid = main_vessel.id.ToString();
-    bool ready_to_draw_active_vessel_trajectory =
-        MapView.MapIsEnabled &&
-        plugin_.HasVessel(main_vessel_guid);
-    if (ready_to_draw_active_vessel_trajectory) {
+    if (MapView.MapIsEnabled) {
       XYZ sun_world_position = (XYZ)Planetarium.fetch.Sun.position;
       using (DisposablePlanetarium planetarium =
                 GLLines.NewPlanetarium(plugin_, sun_world_position)) {
         GLLines.Draw(() => {
+          // Celestial trajectories.
+          foreach (CelestialBody celestial in FlightGlobals.Bodies) {
```

Comment:
This method is already very large and hard to follow.  Please move the celestial-related code to a separate method.